sympy/stats/symbolic_probability.py

```diff
@@ -668,8 +668,6 @@ def __new__(cls, X, n, condition=None, **kwargs):
             return super().__new__(cls, X, n)
 
     def doit(self, **hints):
-        if not is_random(self.args[0]):
-            return self.args[0]
```

Comment:
I guess this should have been something like `self.args[0] ** self.args[1]`.

Should the Moment.doit method be fixed as well?
https://github.com/sympy/sympy/blob/0f6d73ba2a844ce24bb8cd6b8a57177e1bccca23/sympy/stats/symbolic_probability.py#L611-L614

Reply:
Thanks @oscarbenjamin for reviewing. I did the same fix for `moment` and added tests for the n=0 case.